src/vanilla/Model/CodeModelJs.cs

```diff
@@ -49,6 +54,103 @@ public virtual IEnumerable<CompositeTypeJs> OrderedModelTemplateModels
             }
         }
 
+        public virtual string PackageName { get; set; }
+
+        public virtual string PackageVersion { get; set; }
+
+        public virtual string PackageDependencies()
+        {
+            return "\"ms-rest\": \"^2.3.2\"";
+        }
+
+        public string ClientPrefix
+        {
+            get
+            {
+                string clientPrefix = Name;
+
+                const string clientSuffix = "Client";
+                if (clientPrefix.EndsWith(clientSuffix))
+                {
+                    clientPrefix = clientPrefix.Substring(0, clientPrefix.Length - clientSuffix.Length);
+                }
+
+                return clientPrefix;
+            }
+        }
+
+        public string ServiceModelsName => ClientPrefix + "Models";
+
+        public virtual Method GetSampleMethod()
+        {
+            var getMethod = Methods.Where(m => m.HttpMethod == HttpMethod.Get).FirstOrDefault();
+            return getMethod != null ? getMethod : Methods.FirstOrDefault();
+        }
+
+        public virtual string GetSampleMethodGroupName()
+        {
+            return GetSampleMethod()?.MethodGroup?.Name?.ToCamelCase();
+        }
+
+        public virtual string GenerateSampleMethod(bool isBrowser = false)
+        {
+            var method = GetSampleMethod();
+            var methodGroup = GetSampleMethodGroupName();
+            var requiredParameters = method.LogicalParameters.Where(
+                p => p != null && !p.IsClientProperty && !string.IsNullOrWhiteSpace(p.Name) && !p.IsConstant).OrderBy(item => !item.IsRequired).ToList();
+            var builder = new IndentedStringBuilder("  ");
+            var paramInit = InitializeParametersForSampleMethod(requiredParameters, isBrowser);
+            builder.AppendLine(paramInit);
+            var declaration = new StringBuilder();
+            bool first = true;
+            foreach (var param in requiredParameters)
+            {
+                if (!first)
+                    declaration.Append(", ");
+                declaration.Append(param.Name);
+                first = false;
+            }
+            var clientRef = "client.";
+            if (!string.IsNullOrEmpty(methodGroup))
+            {
+                clientRef = $"client.{methodGroup}.";
+            }
+            var methodRef = $"{clientRef}{method.Name.ToCamelCase()}({declaration.ToString()}).then((result) => {{";
+            builder.AppendLine(methodRef)
+                   .Indent()
+                   .AppendLine("console.log(\"The result is:\");")
+                   .AppendLine("console.log(result);")
+                   .Outdent();
+            if (isBrowser)
+            {
+                builder.Append("})");
+            }
+            else
+            {
+                builder.AppendLine("});");

```

Comment:
What's this about?

Reply:
The semicolon versus no semicolon

Reply:
That is used when we're generating the README.md file. It came straight from the TypeScript generator.